logbus/bus.go

```diff
@@ -132,9 +132,11 @@ func (b *Bus) WriteDeltaManifest(dm *logstream.DeltaManifest) {
 			DeltaManifest: dm,
 		},
 	}
+
 	b.rawMu.Lock()
-	defer b.rawMu.Unlock()
 	b.rawHistory = append(b.rawHistory, delta)
+	b.rawMu.Unlock()
```

Comment:
Lock the `append` only, not the cascading `Write`s.

Reply:
If we really don't need to lock for accessing the rawSubs, then perhaps the `rawMu` should be renamed to `rawHistoryMu`?

I don't know the code well enough to know how thread-safety this change is.

Reply:
I think the name is suitable since it's used elsewhere: https://github.com/earthly/earthly/blob/main/logbus/bus.go#L80

Reply:
The change should be thread-safe as there are locks on shared resources in the underlying methods.